lib/mail/renderers/rfc_2822.ex

```diff
@@ -91,7 +93,8 @@ defmodule Mail.Renderers.RFC2822 do
     do: Map.to_list(map)
         |> render_headers()
   def render_headers(list) when is_list(list) do
-    do_render_headers(list)
+    Enum.reject(list, &(Enum.member?(@blacklisted_headers, elem(&1, 0))))
```

Comment:
more optimized since we have a module attribute:

```elixir
Enum.reject(list, fn 
  {header, _val} when header in @blacklisted_headers -> true
  {_header, _val} -> false
end)
```